### PR TITLE
For client v0.15 with app 1.4: protocol fix + address display

### PR DESCRIPTION
**Pull request by cslashm**

### CODE REVIEW

**Comment by selsta** [comment]
Can you add app v1.4 to FORCE_PROVIDER=4 ./Ledger\ Live ?

**Comment by dEBRUYNE-1** [comment]
@ph4r05 - Have you looked at what changes are required for Trezor for 0.15?

**Comment by cslashm** [comment]
@selsta  should be available

File: src/device/device_ledger.cpp [hunk 1]
```diff
@@ -317,6 +353,26 @@ namespace hw {
...
+      offset +=32;
+      if (this->tx_in_progress) {
+        this->hmac_map.find_mac((uint8_t*)sec, this->buffer_send+offset);
+        offset += 32;
+      }
+    }
+
+    void device_ledger::receive_secret(unsigned char sec[32], int &offset) {
+      MDEBUG("receive_secret: " << this->tx_in_progress);
+      memmove(sec, this->buffer_recv+offset, 32);
```
Comment (xiphon) [inline src/device/device_ledger.cpp]: Would be nice to have a bounds check

Comment (xiphon) [reply src/device/device_ledger.cpp]: Same applies to `send_secret` as well

**Comment by moneromooo-monero** [comment]
ping

**Comment by dEBRUYNE-1** [comment]
@cslashm - We will be code freezing / branching soon, please keep in mind.

**Comment by cslashm** [comment]
I m totally out of office until 4 november

**Comment by ph4r05** [comment]
> @ph4r05 - Have you looked at what changes are required for Trezor for 0.15?

btw I've tested Trezor with HF12 and it works.

**Pull request merged**